python/tk_multi_workfiles/browser_form.py

```diff
@@ -705,11 +709,167 @@ def _on_task_status_combo_changed(self, value):
         :param value: The selected value on the combo box.
         """
         if value == "ALL":
+            self._extra_filter = None
             self.my_tasks_model.load_and_refresh()
         else:
-            self.my_tasks_model.load_and_refresh(
-                extra_filter=["sg_status_list", "is", value.lower()]
-            )
+            self._extra_filter = ["sg_status_list", "is", value.lower()]
+            self.my_tasks_model.load_and_refresh(extra_filter=self._extra_filter)
 
         self.my_tasks_model.data_refreshed.emit(True)
-        self.my_tasks_model.async_refresh()
+
+    # Sorting methods
+    def _sort_setup(self):
+        # Last sort menu item selected, by default is sorted 'due_date'
+        self._current_menu_sort_item = ""
+        self._current_menu_sort_order = "asc"
+
+        # Get a task manager from shotgunutils
+        task_manager = sgtk.platform.import_framework(
+            "tk-framework-shotgunutils", "task_manager"
+        )
+        self._task_manager = task_manager.BackgroundTaskManager(
+            self, start_processing=True, max_threads=2
+        )
+
+        # Build an EntityFieldMenu
+        app = sgtk.platform.current_bundle()
+        project_id = app.context.project.get("id")
+
+        self._entity_field_menu = shotgun_menus.EntityFieldMenu(
+            "Task",
+            self,
+            bg_task_manager=self._task_manager,
+            project_id=project_id,
+        )
+
+        # Connect
+        self._my_tasks_form.sort_menu_button.clicked.connect(
+            lambda: self._entity_field_menu.exec_(QtGui.QCursor.pos())
+        )
+
+        # the fields manager is used to query which fields are supported
+        # for display. it can also be used to find out which fields are
+        # visible to the user and editable by the user. the fields manager
+        # needs time to initialize itself. once that's done, the widgets can
+        # begin to be populated.
```

Comment:
```suggestion
        # The fields manager is used to query which fields are supported
        # for display. It can also be used to find out which fields are
        # visible to the user and editable by the user. The fields manager
        # needs time to initialize itself. Once that's done, the widgets can
        # begin to be populated.
```